django/contrib/admin/views/autocomplete.py

```diff
@@ -26,12 +26,21 @@ def get(self, request, *args, **kwargs):
         context = self.get_context_data()
         return JsonResponse({
             'results': [
-                {'id': str(getattr(obj, to_field_name)), 'text': str(obj)}
+                self.obj_to_dict(obj, to_field_name)
                 for obj in context['object_list']
             ],
             'pagination': {'more': context['page_obj'].has_next()},
         })
 
+    def obj_to_dict(self, obj, to_field_name):
```

Comment:
I wonder if something like `serialize_result` might be a more descriptive name for this function.